### PR TITLE
Board: Add support for nucleo_f429zi

### PR DESCRIPTION
**Pull request by erwango**
Add support for board nucleo_f429zi

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

### CODE REVIEW

File: boards/arm/nucleo_f429zi/doc/nucleof429zi.html [hunk 1]
```diff
@@ -0,0 +1,546 @@
+<?xml version="1.0" encoding="utf-8" ?>
```
Comment (dbkinder) [inline boards/arm/nucleo_f429zi/doc/nucleof429zi.html]: Not sure what this .html file is here for, but it shouldn't be here.

File: boards/arm/nucleo_f429zi/doc/nucleof429zi.rst [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+with a wide range of connectivity support and configurations. Here are
+some highlights of the Nucleo F429ZI board:
+
+
+- STM32 microcontroller in LQFP144 package
+- LSE crystal: 32.768 kHz crystal oscillator
+- USB OTG
+- Ethernet compliant with IEEE-802.3-2002
+- Two types of extension resources:
+       - ST Zio connector including: support for Arduino* Uno V3 connectivity
```
Comment (dbkinder) [inline boards/arm/nucleo_f429zi/doc/nucleof429zi.rst]: There should be a blank line before all lists (including this embedded list)

File: boards/arm/nucleo_f429zi/doc/nucleof429zi.rst [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+         (A0 to A5, D0 to D15) and additional signals exposing a wide range of
+         peripherals
+       - ST morpho extension pin headers for full access to all STM32 I/Os
+- On-board ST-LINK/V2-1 debugger/programmer with SWD connector
+- Flexible board power supply:
+       - 5 V from ST-LINK/V2-1 USB VBUS
```
Comment (dbkinder) [inline boards/arm/nucleo_f429zi/doc/nucleof429zi.rst]: blank line before this one too

File: boards/arm/nucleo_f429zi/doc/nucleof429zi.rst [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+
+The default configuration can be found in the defconfig file:
+
+	``boards/arm/nucleo_f429zi/nucleo_f429zi_defconfig``
+
+
+Connections and IOs
+===================
+
+Nucleo F429ZI Board has 8 GPIO controllers. These controllers are responsible for pin muxing,
```
Comment (dbkinder) [inline boards/arm/nucleo_f429zi/doc/nucleof429zi.rst]: The Nucleo ...

File: boards/arm/nucleo_f429zi/doc/nucleof429zi.rst [hunk 5]
```diff
@@ -0,0 +1,181 @@
...
+     :width: 720px
+     :align: center
+     :height: 540px
+     :alt: Nucleo F429ZI Morpho connectors (right)
+
+For mode details please refer to `STM32 Nucleo-144 board User Manual`_.
+
+Default Zephyr Peripheral Mapping:
+----------------------------------
+- UART_3_TX : PD8
```
Comment (dbkinder) [inline boards/arm/nucleo_f429zi/doc/nucleof429zi.rst]: blank like before this list too please.

File: boards/arm/nucleo_f429zi/doc/nucleof429zi.rst [hunk 6]
```diff
@@ -0,0 +1,181 @@
...
+- PWM_2_CH1 : PA0
+- USER_PB : PC13
+- LD1 : PB0
+- LD2 : PB7
+- LD3 : PB14
+
+System Clock
+------------
+
+Nucleo F429ZI System Clock could be driven by internal or external oscillator,
```
Comment (dbkinder) [inline boards/arm/nucleo_f429zi/doc/nucleof429zi.rst]: driven by an internal...

File: boards/arm/nucleo_f429zi/doc/nucleof429zi.rst [hunk 7]
```diff
@@ -0,0 +1,181 @@
...
+as well as main PLL clock. By default System clock is driven by PLL clock at 180MHz,
```
Comment (dbkinder) [inline boards/arm/nucleo_f429zi/doc/nucleof429zi.rst]: as well as by the main...
By default the System clock is driven by the PLL clock...

File: boards/arm/nucleo_f429zi/doc/nucleof429zi.rst [hunk 8]
```diff
@@ -0,0 +1,181 @@
...
+driven by 8MHz high speed external clock.
```
Comment (dbkinder) [inline boards/arm/nucleo_f429zi/doc/nucleof429zi.rst]: driven by an 8MHz...

File: boards/arm/nucleo_f429zi/doc/nucleof429zi.rst [hunk 9]
```diff
@@ -0,0 +1,181 @@
...
+
+Serial Port
+-----------
+
+Nucleo F429ZI board has 8 UARTs. The Zephyr console output is assigned to UART3.
```
Comment (dbkinder) [inline boards/arm/nucleo_f429zi/doc/nucleof429zi.rst]: The Nucleo...

File: boards/arm/nucleo_f429zi/doc/nucleof429zi.rst [hunk 10]
```diff
@@ -0,0 +1,181 @@
...
+Default settings are 115200 8N1.
+
+
+Programming and Debugging
+*************************
+
+Nucleo F429ZI board includes an ST-LINK/V2-1 embedded debug tool interface.
```
Comment (dbkinder) [inline boards/arm/nucleo_f429zi/doc/nucleof429zi.rst]: The Nucleo...

**Review by dbkinder — CHANGES_REQUESTED** [review]
suggested edits. Also, there's an html file here that doesn't belong.

**Comment by erwango** [comment]
@dbkinder : Thanks for the review. Should be ok now.

**Review by galak — APPROVED** [review]
Looks good!

**Comment by galak** [comment]
@dbkinder you good with the changes?

**Review by dbkinder — APPROVED** [review]
thanks!

**Pull request merged**